File: sentimental/readability/readability.py
from cs50 import get_string
# Prompt for Text
txt = get_string("Text: ")
# Define total length of Text, total letters, spaces and sentences
l = len(txt)
letters = 0
spaces = 0
sentences = 0
# Calculate total letters, spaces and sentences in Text
for i in range(l):
    # Take all upper-case and lower-case chars as letters
    if txt[i].islower():
        letters += 1
    elif txt[i].isupper():
        letters += 1
    # Count spaces
    elif txt[i] == " ":
        spaces += 1
    # Search for different periods to calculate total sentences
    elif txt[i] == "." or txt[i] == "!" or txt[i] == "?":
        sentences += 1
# Calculate total words in Text
words = spaces + 1
# Calculate the Average number of letters per 100 words in Text
L = (letters * 100) / words
# Calculate the Average number of sentences per 100 words in Text
S = (sentences * 100) / words
# Calculate the The Coleman-Liau Index of the Text and ROUND IT TO THE NEAREST WHOLE NUMBER
index = round(0.0588 * L - 0.296 * S - 15.8)
# Different conditions of index / Grade
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(index))

File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <string.h>
#include <stdio.h>

int main(void)
{
    //Prompt for Text
    string text = get_string("Text: ");
    //Define total length of Text, total Letters, Spaces and Sentences
    int len = strlen(text), letters = 0, spaces = 0, sentences = 0;
    //Calculate total Letters, Spaces and Sentences in Text
    for(int i = 0; i < len; i++)
    {
        //Take all upper-case and lower-case chars as Letters
        if(islower(text[i]))
        {
            letters++;
        }
        else if(isupper(text[i]))
        {
            letters++;
        }
        else if(text[i] == ' ')
        {
            spaces++;
        }
        //Search for different periods to calculate total sentences
        else if(text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    //Calculate total Words in Text
    int words = spaces + 1;
    //Calculate the Average number of Letters per 100 Words in Text
    float L = (float)letters * (float)100 / (float)words;
    //Calculate the Average number of Sentences per 100 Words in Text
    float S = (float)sentences * (float)100 / (float)words;
    //Calculate the The Coleman-Liau Index of the Text and ROUND IT TO THE NEAREST WHOLE NUMBER
    float index = round(0.0588 * L - 0.296 * S - 15.8);
    //Define Grade
    int grade = index;
    //Different conditions of Grade
    if(grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if(grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}